internal/eval/eval.go

```diff
@@ -675,6 +675,162 @@ func evalJSONSET(args []string, store *dstore.Store) []byte {
 	return clientio.RespOK
 }
 
+// evalJSONARRAPPEND appends the value(s) provided in the args to the given array path
+// in the JSON object saved at key in arguments.
+// Args must contain atleast a key, path and value.
+// If the key does not exist or is expired, it returns response.RespNIL.
+// If the object at given path is not an array, it returns response.RespNIL.
+// Returns the new length of the array at path.
+func evalJSONARRAPPEND(args []string, store *dstore.Store) []byte {
```

Comment:
I feel this implementation can be simplified a lot. I tried coding it on my machine and I think you can also do something similar here. I tested it and it works fine for the most part. It also allows you to have arrays of mixed data types (while maintaining type information)

Note that this is just a rough implementation, please match the error messages exactly with how redis does them (and add tests for each one):

```go
func evalJSONARRAPPEND(args []string, store *dstore.Store) []byte {
	if len(args) < 3 {
		return diceerrors.NewErrArity("JSON.ARRAPPEND")
	}

	key := args[0]
	path := args[1]
	values := args[2:]

	obj := store.Get(key)
	if obj == nil {
		return diceerrors.NewErrWithMessage("ERR key does not exist")
	}

	err := object.AssertTypeAndEncoding(obj.TypeEncoding, object.ObjTypeJSON, object.ObjEncodingJSON)
	if err != nil {
		return err
	}

	jsonData := obj.Value

	expr, parseErr := jp.ParseString(path)
	if parseErr != nil {
		return diceerrors.NewErrWithMessage("ERR invalid JSONPath")
	}

	// Parse the input values as JSON
	parsedValues := make([]interface{}, len(values))
	for i, v := range values {
		var parsedValue interface{}
		err := sonic.UnmarshalString(v, &parsedValue)
		if err != nil {
			return diceerrors.NewErrWithMessage(fmt.Sprintf("ERR invalid JSON value: %s", v))
		}
		parsedValues[i] = parsedValue
	}

	var resultsArray []interface{}
	modified := false

	// Capture the modified data when modifying the root path
	var newData interface{}
	var modifyErr error

	newData, modifyErr = expr.Modify(jsonData, func(data any) (interface{}, bool) {
		arr, ok := data.([]interface{})
		if !ok {
			// Not an array
			resultsArray = append(resultsArray, nil)
			return data, false
		}

		// Append the parsed values to the array
		arr = append(arr, parsedValues...)

		// Record the new length
		resultsArray = append(resultsArray, int64(len(arr)))

		modified = true

		return arr, modified
	})

	if modifyErr != nil {
		return diceerrors.NewErrWithMessage(fmt.Sprintf("ERR failed to modify JSON data: %v", modifyErr))
	}

	if !modified {
		// If no modification was made, it means the path did not exist or was not an array
		return clientio.Encode([]interface{}{nil}, false)
	}

	jsonData = newData

	obj.Value = jsonData

	return clientio.Encode(resultsArray, false)
}
```